lib.php

```diff
@@ -214,15 +209,27 @@ public function get_links($linkarray) {
             // Get turnitin submission details.
             $plagiarismfile = plagiarism_turnitinsim_submission::get_submission_details($linkarray);
 
+            $submission = null;
+
             // The links for forum posts get shown to all users.
             // Return if the logged in user shouldn't see OR scores. E.g. forum posts.
             if (!$moduleobject->show_other_posts_links($cm->course, $linkarray['userid'])) {
                 return $output;
             }
 
-            // Render the OR score or current submission status.
             if ($plagiarismfile) {
                 $submission = new plagiarism_turnitinsim_submission(new plagiarism_turnitinsim_request(), $plagiarismfile->id);
+            }
+
+            // If the user is a student and they are not allowed to view reports then return empty output.
```

Comment:
If I was being a pedant I would say expand on that comment but it's a pretty readable if statement.